crates/nu-cmd-lang/src/core_commands/collect.rs

```diff
@@ -88,7 +101,7 @@ impl Command for Collect {
             },
             Example {
                 description: "Read and write to the same file",
-                example: "open file.txt | collect { save -f file.txt }",
+                example: "open file.txt | collect | save -f file.txt",
```

Comment:
... I suspect this will fail now (as previously it was running the closure where the metadata was not set)

Reply:
![grafik](https://github.com/nushell/nushell/assets/15833959/cf8ae2be-1b4d-48fc-87cb-ac45b1f13dff)

Reply:
I think we could maybe remove the same file check in `save` for `PipelineData::Value`. Also, perhaps we should just automatically do a `collect` in `save` if we detect the error condition?

Reply:
Ignoring it for `PipelineData::Value` sounds like a good idea (until we have lazy data in `Value` coming from a file)

Not sure if I would want to introduce magic into `save` to collect just yet. There may be cases where it alerts you to otherwise bad ideas like round tripping on an otherwise never terminating file/socket etc.?

Reply:
Makes sense. I actually assumed that was already the case